scripts/new_upstream_snapshot.py

```diff
@@ -350,9 +367,7 @@ def update_changelog(
     """
     print("Updating changelog")
     old_version = changelog_details.version
-    previously_unreleased = (
-        changelog_details.distro.upper() == "UNRELEASED"
-    )
+    previously_unreleased = changelog_details.distro.upper() == "UNRELEASED"
     commitish_is_upstream_tag = is_commitish_upstream_tag(commitish)
     old_series_suffix = get_series_suffix(old_version)
```

Comment:
This function breaks on current devel release versioning as it sees the `~` in `23.3~1ge5a617fe-0ubuntu1` and returns the old_series_suffix `1ge5a617fe-0ubuntu1`. I think we need to adapt this function too to ensure we are properly parsing the ~\d pre-release versioning scheme